### ISSUE TITLE
Mention with HTML tags aren't escaping

### ISSUE THREAD

**Issue by handonam** [angular-ui/ui-mention#20]
When you start typing something such as `<a href="#">hello`, the rendered text starts to generate the anchor tag.

![screen shot 2016-03-05 at 2 18 27 pm](https://cloud.githubusercontent.com/assets/455463/13550901/4a54e8ee-e2dd-11e5-892a-aa533a6d9f2c.png)

This should only be escaping our `@mention` use cases, and not for raw HTML such as anchors.  This also seems to happen on other tags as `<em>`, `<strong>`, and a few others that may be going into `$element.next().html(html);` in render (https://github.com/angular-ui/ui-mention/blob/master/src/mentionController.es6.js#L75)

You can see the text being italicized/bolded if you remove this rule from the highlight div: `color: transparent`

**Comment by handonam** [comment]
Could this possibly work?

``` javascript
 this.render = function () {
    var html = arguments.length <= 0 || arguments[0] === undefined ? ngModel.$modelValue : arguments[0];

    html = (html || '').toString();

+    var entityMap = {
+      "&": "&amp;",
+      "<": "&lt;",
+      ">": "&gt;",
+      '"': '&quot;',
+      "'": '&#39;',
+      "/": '&#x2F;'
+    };
+    
+    // escape these html characters into entities
+    html = String(html).replace(/[&<>"'\/]/g, function (s) {
+      return entityMap[s];
+    });
+
+    // transform the mentions and wrap them into unescaped `<span>` tags
    _this2.mentions.forEach(function (mention) {
      html = html.replace(_this2.encode(mention), _this2.highlight(mention));
    });
    $element.next().html(html);
    return html;
  };
```

the escaping function was pulled out of mustache.js (https://github.com/janl/mustache.js/blob/master/mustache.js#L82) as I was reading this (http://stackoverflow.com/questions/14462612/escape-html-text-in-an-angularjs-directive/28537958#28537958)

**Comment by ProLoser** [comment]
Have you tried hotpatching the render function in your wrapper directive?

**Comment by ProLoser** [comment]
Like this:

``` js
var render = uiMention.render.bind(uiMention);
uiMention.render = function(html) {
  html = html || ngModel.$modelValue;
  if (typeof html == 'string')
    html = replace(html);
  render(html);
}
```

**Comment by handonam** [comment]
hm... so I tried this:

``` js
        var render = uiMention.render.bind(uiMention);
        uiMention.render = function(html) {
          html = html || uiMention.$modelValue;

          var entityMap = {
            "&": "&amp;",
            "<": "&lt;",
            ">": "&gt;",
            '"': '&quot;',
            "'": '&#39;',
            "/": '&#x2F;'
          };

          html = String(html).replace(/[&<>"'\/]/g, function(s) {
            return entityMap[s];
          });

          render(html);
        }
```

however, it won't render the mentions until i type another character.  This leads me to believe that the hotpatching occurs AFTER the base render method.

edit: no, i was wrong. it's because there's an `ngModel.render()` for the textArea that runs through this process again.  my `uiMention.$modelValue` wasn't the right choice to get the model value

edit 2:  Okay, sorry for the edits.  This is the final solution that worked for me.  I would still consider it a warning at the very least when typing HTML code in to a directive that will end up straight into an element as HTML.

``` js
        var render = uiMention.render.bind(uiMention);
        uiMention.render = function(html) {
          html = html || $scope[$attrs.ngModel];

          var entityMap = {
            "&": "&amp;",
            "<": "&lt;",
            ">": "&gt;",
            '"': '&quot;',
            "'": '&#39;',
            "/": '&#x2F;'
          };

          html = String(html).replace(/[&<>"'\/]/g, function(s) {
            return entityMap[s];
          });

          render(html);
        }
```

**Comment by ProLoser** [comment]
Hmm yeah since anyone could type in HTML this should probably be fixed in core.

**Comment by handonam** [comment]
I can make a PR for this, it would be the first codeblock I mentioned above.

**Comment by ProLoser** [comment]
yeah

**Comment by JacobMarshallPP** [comment]
Just noticed this issue, I just submitted a PR with a fix for this. See #26

**Comment by ProLoser** [comment]
Fixed by #26
